src/overlays/actors/ovl_Obj_Kibako/z_obj_kibako.c

```diff
@@ -27,33 +37,242 @@ const ActorInit Obj_Kibako_InitVars = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 60, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+void ObjKibako_SpawnCollectible(ObjKibako* this, GlobalContext* globalCtx) {
+    s16 collectible;
+
+    collectible = this->actor.params & 0x1F;
+    if ((collectible >= 0) && (collectible <= 0x19)) {
+        Item_DropCollectible(globalCtx, &this->actor.posRot.pos,
+                             collectible | (((this->actor.params >> 8) & 0x3F) << 8));
+    }
+}
+
+void ObjKibako_ApplyGravity(ObjKibako* this) {
+    this->actor.velocity.y += this->actor.gravity;
+    if (this->actor.velocity.y < this->actor.minVelocityY) {
+        this->actor.velocity.y = this->actor.minVelocityY;
+    }
+}
+
+void ObjKibako_InitCollider(Actor* thisx, GlobalContext* globalCtx) {
+    ObjKibako* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    Collider_CylinderUpdate(&this->actor, &this->collider);
+}
+
+void ObjKibako_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjKibako* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->actor.gravity = -1.2f;
+    this->actor.minVelocityY = -13.0f;
+    ObjKibako_InitCollider(&this->actor, globalCtx);
+    func_80061ED4(&this->actor.colChkInfo, NULL, &sCCInfoInit);
+    ObjKibako_SetupIdle(this);
+    // wooden box
+    osSyncPrintf("(dungeon keep 木箱)(arg_data 0x%04x)\n", this->actor.params);
+}
+
+void ObjKibako_Destroy(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    ObjKibako* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void ObjKibako_AirBreak(ObjKibako* this, GlobalContext* globalCtx) {
+    s16 angle;
+    s32 i;
+    Vec3f* pos = &this->actor.posRot.pos;
+    Vec3f spC8;
+    Vec3f spBC;
+
+    for (i = 0, angle = 0; i < 12; i++, angle += 0x4E20) {
+        f32 sn = Math_Sins(angle);
+        f32 cs = Math_Coss(angle);
+        f32 temp_rand;
+        s16 phi_s0;
+
+        spC8.x = sn * 16.0f;
+        spC8.y = (Math_Rand_ZeroOne() * 5.0f) + 2.0f;
+        spC8.z = cs * 16.0f;
+        spBC.x = spC8.x * 0.2f;
+        spBC.y = (Math_Rand_ZeroOne() * 6.0f) + 2.0f;
+        spBC.z = spC8.z * 0.2f;
+        spC8.x += pos->x;
+        spC8.y += pos->y;
+        spC8.z += pos->z;
+        temp_rand = Math_Rand_ZeroOne();
+        if (temp_rand < 0.1f) {
+            phi_s0 = 0x60;
+        } else if (temp_rand < 0.7f) {
+            phi_s0 = 0x40;
+        } else {
+            phi_s0 = 0x20;
+        }
+        EffectSsKakera_Spawn(globalCtx, &spC8, &spBC, pos, -200, phi_s0, 10, 10, 0,
+                             (Math_Rand_ZeroOne() * 30.0f) + 10.0f, 0, 32, 60, KAKERA_COLOR_NONE,
+                             OBJECT_GAMEPLAY_DANGEON_KEEP, D_05005380);
```

Comment:
spC8 is pos and spBC is velocity
i would rename what you currently have as `pos` to boxPos or something, since its not the pos of the fragment being spawned

same for the usage below